src/luarocks/util.lua

```diff
@@ -373,6 +353,21 @@ function util.split_namespace(ns_name)
    return ns_name
 end
 
+--- Argparse action callback for namespaced rock arguments.
+function util.namespaced_name_action(args, target, ns_name)
```

Comment:
This looks a lot like `adjust_name_and_namespace`: could you modify the behaviour of that function?
Also e.g. keep the `assert`s?